core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -111,6 +102,17 @@ public class SplittingTransformer<ResponseT, ResultT> implements SdkPublisher<As
 
     private final Object cancelLock = new Object();
 
+    /**
+     * Keeps track of the upstream future returned by {@code upstreamResponseTransformer.prepare()}. If an error occurs, we
+     * forward this to the {@code resultFuture}.
+     */
+    private volatile CompletableFuture<ResultT> upstreamFuture;
+
+    /**
+     * Tracks the part number. Errors will only be retried for the first part.
+     */
+    private final AtomicInteger partNumber = new AtomicInteger(0);
```

Comment:
NIT (non-blocking): I think we should avoid referring to 'parts' in the `SplittingTranformer`, that's a s3 concept, when this class is more abstract, in a core package. Maybe rename to something like 'onNextSignalsSent' or `onNextNumber` which would track the total amount of 'onNext' signals sent to the `downstreamSubscriber`